File: app/src/components/HaftaSecici.js
// HAFTA SEÇİCİ — Radar Merkezi'nin üstündeki hafta gezintisi.
//
// ESKİ HÂLİ: bütün haftalar yan yana çipti ("53 · Güncel | 52 🔏 | 51 🔏").
// Yeni sezon 1. haftayla başlayınca hem numaralar küçülüyor hem de haftalar
// birikiyor (sezonda 52) — şerit okunmaz hâle geliyor.
//
// YENİ KALIP resmî Spor Toto listesindekiyle aynıdır — İKİ açılır liste:
//
//   [2025/2026 Sezonu ▼]   [53. Hafta ▼]
//
// Hafta listesi seçili sezonun BÜTÜN haftalarını taşır, güncel dahil, en
// üstte. Sezon TEK ise açılır liste değil DÜZ YAZI gösterilir: tek seçenekli
// bir liste, dokunup hiçbir şeyin değişmemesi demektir.
//
// Bu bileşen DURUM TUTMAZ — açık/kapalı ve seçimler RadarScreen'de yaşar.
import React from 'react';
import { View, Text, ScrollView, TouchableOpacity, StyleSheet } from 'react-native';
import { colors, radius } from '../theme';
import { haftaSeciciVerisi } from '../radarScreenData';

// Liste bu yükseklikten sonra kaydırılır: 52 haftalık sezon tek ekrana sığmaz
// (resmî listede de kaydırma vardır).
const LISTE_MAX_YUKSEKLIK = 300;

export default function HaftaSecici({
  weeks, curId, selectedId,
  acik = null,                    // null | 'sezon' | 'hafta' — tek seferde tek liste
  onToggle = () => {},
  navSezon = null, onSelectSezon = () => {},
  onSelectWeek = () => {},
}) {
  const v = haftaSeciciVerisi(weeks, { curId, selectedId, navSezon });
  if (!v.liste.length) return null;

  const cokSezon = v.sezonlar.length > 1;

  return (
    <View>
      <View style={styles.satir}>
        {cokSezon ? (
          <TouchableOpacity
            onPress={() => onToggle('sezon')}
            activeOpacity={0.75}
            accessibilityRole="button"
            accessibilityLabel={`Sezon seç, şu an ${v.sezonAdi}`}
            style={styles.sec}
          >
            <Text style={styles.secTxt}>{v.sezonAdi}</Text>
            <Text testID="sezon-ok" style={styles.ok}>{acik === 'sezon' ? '⌃' : '⌄'}</Text>
          </TouchableOpacity>
        ) : (
          // Tek sezon: seçecek bir şey yok, düz yazı.
          <Text style={styles.tekSezon}>{v.sezonAdi}</Text>
        )}

        <TouchableOpacity
          onPress={() => onToggle('hafta')}
          activeOpacity={0.75}
          accessibilityRole="button"
          accessibilityLabel={`Hafta seç, şu an ${v.haftaAdi || 'seçili değil'}`}
          style={styles.sec}
        >
          <Text style={styles.secTxt}>
            {v.haftaAdi || 'Seç'}{v.haftaGuncelMi ? ' · Güncel' : ''}
          </Text>
          <Text testID="hafta-ok" style={styles.ok}>{acik === 'hafta' ? '⌃' : '⌄'}</Text>
        </TouchableOpacity>
      </View>

      {acik === 'sezon' && cokSezon ? (
        <View style={styles.liste}>
          {v.sezonlar.map((s) => {
            const secili = s.y === v.seciliSezon;
            return (
              <TouchableOpacity
                key={s.y}
                style={[styles.oge, secili && styles.ogeSecili]}
                onPress={() => onSelectSezon(s.y)}
                activeOpacity={0.8}
              >
                <Text style={[styles.ogeTxt, secili && styles.ogeTxtSecili]}>{s.ad}</Text>
              </TouchableOpacity>
            );
          })}
        </View>
      ) : null}

      {acik === 'hafta' ? (
        <View style={[styles.liste, styles.listeKaydirmali]}>
          <ScrollView nestedScrollEnabled showsVerticalScrollIndicator>
            {v.liste.map((w) => {
              const secili = Number(w.roundId) === Number(selectedId);
              return (
                <TouchableOpacity
                  key={w.roundId}
                  style={[styles.oge, secili && styles.ogeSecili]}
                  onPress={() => onSelectWeek(w.roundId)}
                  activeOpacity={0.8}
                >
                  <Text style={[styles.ogeTxt, secili && styles.ogeTxtSecili]}>{w.ad}</Text>
                  {/* Sağdaki işaret: güncel mi, mühürlü mü. Mühür "sonradan
                      değişmez" güvencesidir, gizlenmez. */}
                  <Text style={styles.ogeMeta}>
                    {w.guncel ? 'Güncel' : w.kilitli ? '🔏' : ''}
                  </Text>
                </TouchableOpacity>
              );
            })}
          </ScrollView>
        </View>
      ) : null}
    </View>
  );
}

const styles = StyleSheet.create({
  satir: { flexDirection: 'row', flexWrap: 'wrap', alignItems: 'center', gap: 14 },
  // Resmî listedeki gibi sade: çerçevesiz yazı + küçük ok.
  sec: { flexDirection: 'row', alignItems: 'center', gap: 6, paddingVertical: 4 },
  secTxt: { color: colors.text, fontSize: 14, fontWeight: '700' },
  tekSezon: { color: colors.textSoft, fontSize: 14, fontWeight: '700', paddingVertical: 4 },
  ok: { color: colors.muted, fontSize: 13, fontWeight: '900' },
  liste: {
    marginTop: 6, borderWidth: 1, borderColor: colors.border, borderRadius: radius.sm,
    backgroundColor: colors.card, overflow: 'hidden', alignSelf: 'flex-start', minWidth: 210,
  },
  listeKaydirmali: { maxHeight: LISTE_MAX_YUKSEKLIK },
  oge: {
    flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between', gap: 12,
    paddingVertical: 11, paddingHorizontal: 16,
    borderBottomWidth: 1, borderBottomColor: colors.border,
  },
  ogeSecili: { backgroundColor: colors.primarySoft },
  ogeTxt: { color: colors.primary, fontSize: 13.5, fontWeight: '700' },
  ogeTxtSecili: { color: colors.primary, fontWeight: '900' },
  ogeMeta: { color: colors.muted, fontSize: 11, fontWeight: '700' },
});

File: lib/features/radar/hafta_secici.dart
// KAYNAK: app/src/components/HaftaSecici.js — BİREBİR çeviri.
//
// HAFTA SEÇİCİ — Radar Merkezi'nin üstündeki hafta gezintisi.
//
// ESKİ HÂLİ: bütün haftalar yan yana çipti ("53 · Güncel | 52 🔏 | 51 🔏").
// Yeni sezon 1. haftayla başlayınca hem numaralar küçülüyor hem de haftalar
// birikiyor (sezonda 52) — şerit okunmaz hâle geliyor.
//
// YENİ KALIP resmî Spor Toto listesindekiyle aynıdır — İKİ açılır liste:
//
//   [2025/2026 Sezonu ▼]   [53. Hafta ▼]
//
// Hafta listesi seçili sezonun BÜTÜN haftalarını taşır, güncel dahil, en
// üstte. SEZON tek olsa bile AÇILIR kalır: düz yazı denendi ve "açılır olduğu
// anlaşılmıyor" geri bildirimi geldi; ikinci sezon gelince görünümün
// değişmemesi de daha tutarlı.
//
// Bu bileşen DURUM TUTMAZ — açık/kapalı ve seçimler RadarScreen'de yaşar.

import 'package:flutter/material.dart';

import '../../core/theme/tokens.dart';
import 'radar_screen_data.dart';

/// Liste bu yükseklikten sonra kaydırılır: 52 haftalık sezon tek ekrana
/// sığmaz (resmî listede de kaydırma vardır).
const double _kListeMaxYukseklik = 300;

class HaftaSecici extends StatelessWidget {
  const HaftaSecici({
    super.key,
    required this.weeks,
    required this.curId,
    required this.selectedId,
    this.acik, // null | 'sezon' | 'hafta' — tek seferde tek liste
    required this.onToggle,
    this.navSezon,
    required this.onSelectSezon,
    required this.onSelectWeek,
    this.onDisariTiklandi,
  });

  final List? weeks;
  final Object? curId;
  final Object? selectedId;
  final String? acik;
  final ValueChanged<String> onToggle;
  final Object? navSezon;
  final ValueChanged<String> onSelectSezon;
  final ValueChanged<int> onSelectWeek;

  /// Açık listeye DIŞARI tıklanınca kapanma (2026-08-10, "profesyonel
  /// uygulama" turu): açılır liste yalnız düğmeyle değil, ekranın başka
  /// yerine dokununca da kapanmalı — standart açılır liste davranışı.
  final VoidCallback? onDisariTiklandi;

  @override
  Widget build(BuildContext context) {
    final v = haftaSeciciVerisi(
      weeks,
      curId: curId,
      selectedId: selectedId,
      navSezon: navSezon,
    );
    if (v.liste.isEmpty) return const SizedBox.shrink();

    // LİSTE KENDİ DÜĞMESİNİN ALTINDA AÇILIR (kullanıcı bildirimi, 2026-08-10:
    // "geçmiş haftalar solda çıkıyor, güncel haftanın altında çıkmalı").
    // Eskiden iki liste de satırın SOL kenarına iniyordu; sezon değiştirince
    // açılan HAFTA listesi sezonun listesi gibi görünüyordu. Her düğme kendi
    // sütununda: listesi tam altına iner, dar ekranda Wrap alt satıra taşır.
    return TapRegion(
      onTapOutside: (_) {
        if (acik != null) onDisariTiklandi?.call();
      },
      child: _govde(v),
    );
  }

  Widget _govde(HaftaSeciciVerisi v) {
    return Wrap(
      spacing: 8,
      runSpacing: 8,
      children: [
        Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            // SEZON — tek sezon olsa bile AÇILIR kalır (üstteki not).
            Semantics(
              button: true,
              label: 'Sezon seç, şu an ${v.sezonAdi}',
              child: _secDugmesi(
                metin: v.sezonAdi,
                okKey: const Key('sezon-ok'),
                acikMi: acik == 'sezon',
                onTap: () => onToggle('sezon'),
              ),
            ),
            if (acik == 'sezon')
              _liste(
                kaydirmali: false,
                children: [
                  for (final s in v.sezonlar)
                    _oge(
                      metin: s.ad,
                      meta: '',
                      secili: s.y == v.seciliSezon,
                      onTap: () => onSelectSezon(s.y),
                    ),
                ],
              ),
          ],
        ),
        Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Semantics(
              button: true,
              label: 'Hafta seç, şu an ${v.haftaAdi ?? 'seçili değil'}',
              child: _secDugmesi(
                metin:
                    '${v.haftaAdi ?? 'Seç'}${v.haftaGuncelMi ? ' · Güncel' : ''}',
                okKey: const Key('hafta-ok'),
                acikMi: acik == 'hafta',
                onTap: () => onToggle('hafta'),
              ),
            ),
            if (acik == 'hafta')
              _liste(
                kaydirmali: true,
                children: [
                  for (final w in v.liste)
                    _oge(
                      metin: w.ad,
                      // Sağdaki işaret: güncel mi, mühürlü mü. Mühür "sonradan
                      // değişmez" güvencesidir, gizlenmez.
                      meta: w.guncel ? 'Güncel' : (w.kilitli ? 'Kilitli' : ''),
                      secili:
                          selectedId != null &&
                          w.roundId == num.tryParse('$selectedId'),
                      onTap: () => onSelectWeek(w.roundId),
                    ),
                ],
              ),
          ],
        ),
      ],
    );
  }

  // DOKUNULABİLİR OLDUĞU BELLİ OLMALI: çerçeve + zemin + ok. Önce resmî
  // listedeki gibi çerçevesiz düz yazı denendi, "açılır olduğu anlaşılmıyor"
  // geri bildirimi geldi.
  Widget _secDugmesi({
    required String metin,
    required Key okKey,
    required bool acikMi,
    required VoidCallback onTap,
  }) => GestureDetector(
    onTap: onTap,
    behavior: HitTestBehavior.opaque,
    child: Container(
      padding: const EdgeInsets.symmetric(horizontal: 12, vertical: 7),
      decoration: BoxDecoration(
        color: acikMi ? AppColors.primarySoft : AppColors.surfaceSoft,
        borderRadius: AppRadius.pillR,
        border: Border.all(
          color: acikMi ? AppColors.primary : AppColors.border,
        ),
      ),
      child: Row(
        mainAxisSize: MainAxisSize.min,
        children: [
          Text(
            metin,
            style: TextStyle(
              color: AppColors.text,
              fontSize: 13.5,
              fontWeight: AppFont.heavy,
            ),
          ),
          const SizedBox(width: 8),
          // ▼/▲ (U+25BC/25B2) — önce ⌄/⌃ (U+2304/2303) kullanılmıştı ve
          // cihazda GÖRÜNMÜYORDU; o karakterler birçok yazı tipinde yok. Bu
          // ikisi aynı ekranda Radar 5 satır açılımında zaten çalışıyor.
          Text(
            acikMi ? '▲' : '▼',
            key: okKey,
            style: TextStyle(
              color: AppColors.primary,
              fontSize: 11,
              fontWeight: AppFont.black,
            ),
          ),
        ],
      ),
    ),
  );

  Widget _liste({required bool kaydirmali, required List<Widget> children}) {
    final govde = Column(mainAxisSize: MainAxisSize.min, children: children);
    return Container(
      margin: const EdgeInsets.only(top: 6),
      constraints: BoxConstraints(
        minWidth: 210,
        maxHeight: kaydirmali ? _kListeMaxYukseklik : double.infinity,
      ),
      decoration: BoxDecoration(
        color: AppColors.card,
        borderRadius: AppRadius.smR,
        border: Border.all(color: AppColors.border),
      ),
      clipBehavior: Clip.hardEdge,
      child: kaydirmali ? SingleChildScrollView(child: govde) : govde,
    );
  }

  Widget _oge({
    required String metin,
    required String meta,
    required bool secili,
    required VoidCallback onTap,
  }) => GestureDetector(
    onTap: onTap,
    behavior: HitTestBehavior.opaque,
    child: Container(
      padding: const EdgeInsets.symmetric(vertical: 11, horizontal: 16),
      decoration: BoxDecoration(
        color: secili ? AppColors.primarySoft : Colors.transparent,
        border: Border(bottom: BorderSide(color: AppColors.border)),
      ),
      child: Row(
        mainAxisSize: MainAxisSize.min,
        children: [
          Text(
            metin,
            style: TextStyle(
              color: AppColors.primary,
              fontSize: 13.5,
              fontWeight: secili ? AppFont.black : AppFont.bold,
            ),
          ),
          const SizedBox(width: 12),
          Text(
            meta,
            style: TextStyle(
              color: AppColors.textMuted,
              fontSize: 11,
              fontWeight: AppFont.bold,
            ),
          ),
        ],
      ),
    ),
  );
}
